src/Filter.php

```diff
@@ -27,16 +27,16 @@ class Filter
     public function apply(string $bem, string $filter): string
     {
         // Allow to be used outside of WordPress.
-        if (\function_exists('apply_filters')) {
+        if (\function_exists('apply_filters_ref_array')) {
             /**
              * Bem Filter
              *
              * Filter the value string before it is returned.
              *
-             * @param string $bem The bem value.
-             * @param Valuable $this The instance of the class.
+             * @param string $filter The hook name.
+             * @param array{string, Valuable} $ref_array The bem value and the instance of this class.
              */
-            $bem = (string)apply_filters($filter, $bem, $this);
+            $bem = (string)\apply_filters_ref_array($filter, [$bem, $this]);
```

Comment:
Interested in understanding why this would be a better option instead of passing the parameters?

Reply:
WordPress core passes `$this` only in arrays.
Please run `grep -rn 'apply_filters.*\$this'` on the core.

Reply:
Sounds like the problem is in the call to `func_get_args` see https://3v4l.org/TpQAg `apply_filters` get passed only one argument along with the tag and all others are retrieved dinamically by calling `func_get_args` it will return a copy of the value passed to the function.
Thanks for pointing out that I didn't know that.